specs/NonClientHitTestKind.md

```diff
@@ -0,0 +1,182 @@
+        // to webview
+        isCapturingMouse = false 
+        ReleaseCapture();
+    }
+
+    // forward mouse messages to WV
+    CHECK_FAILURE(CompositionController->SendMouseInput(
+        static_cast<COREWEBVIEW2_MOUSE_EVENT_KIND>(message),
+        static_cast<COREWEBVIEW2_MOUSE_EVENT_VIRTUAL_KEYS>(GET_KEYSTATE_WPARAM(wParam)),
+        mouseData, point));
+    return 0;
+    }
+    
+}
+// The following code forwards mouse messages to the WebView2 - End
+
+switch(message) 
+{
+    ....
+    // Handling this message is important for enabling dragging
+    case WM_NCHITTEST:
+    POINT point {GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam)};
+    COREWEBVIEW2_HIT_TEST_KIND kind = NULL;
+    CompositionController->NonClientHitTestPoint(&point, &kind);
+
+    if (kind == 
+        COREWEBVIEW2_NON_CLIENT_HIT_TEST_KIND::COREWEBVIEW2_NON_CLIENT_HIT_TEST_KIND_HTCAPTION)
+    {
+        return HTCAPTION;
+    }
+
+    break;
+    ....
+}
+....
+}
+```
+#### C#/WinRT
+```c#
+protected override void WndProc(ref Message m)
+{
+....
+
+bool isCapturingMouse = false;
+
+if (m.Msg == WM_NCHITTEST)
+{
+    Point point = new Point(GET_X_LPARAM(m.lParam), GET_Y_LPARAM(m.lParam));
+    if (CompositionController.NonClientHitTestPoint(point) == 
+        CoreWebView2NonClientHitTestKind.Htcaption) 
+    {
+        m.Result = (IntPtr)HTCAPTION;
+    }
+}
+// The following code forwards mouse messages to the WebView2 - Start
+else if (m.msg >= WM_MOUSEFIRST && m.msg <= WM_MOUSELAST) 
+{
+  
+    Point point = new Point(GET_X_LPARAM(m.lParam), GET_Y_LPARAM(m.lParam));
+    // IsPointInWebview is an example of a function which your app can implement that checks if 
+    // the mouse message point lies inside the rect of the visual thosts the WebView
+    if (IsPointInWebView(point) || m.msg == WM_MOUSELEAVE || isCapturingMouse) 
+    {
+
+        if (m.msg == WM_NCRBUTTONDOWN && m.w_param == HTCAPTION) 
+        {
+            // capturing the mouse will allow us to begin forwarding messages 
+            // to webview
+            isCapturingMouse = true;
+            SetCapture(mainWindowHwnd);
+        }
+        else if (message ==  WM_RBUTTONUP && GetCapture() == mainWindowHwnd) 
+        {
+            // no longer need to capture the mouse for draggbale regions support 
+            // after this
+            isCapturingMouse = false;
+            ReleaseCapture();
+        }
+        
+        // forward mouse messages to WV
+        CompositionController.SendMouseInput(
+            (COREWEBVIEW2_MOUSE_EVENT_KIND)message,
+            (COREWEBVIEW2_MOUSE_EVENT_VIRTUAL_KEYS)GET_KEYSTATE_WPARAM(wParam),
+            mouseData,
+            point);
+
+        m.Result = 0
+        return;
+    }
+        
+}
+// The following code forwards mouse messages to the WebView2 - Start
+    ....
+}
+       
+```
+# API Details
+## Win32 C++
+```cpp
+typedef enum COREWEBVIEW2_NON_CLIENT_HIT_TEST_KIND {
+  COREWEBVIEW2_NON_CLIENT_HIT_TEST_KIND_HTCAPTION,
```

Comment:
Does the HT in HTCAPTION stand for Hit Test? Its sort of repetitive if so? Should it just be ...HIT_TEST_KIND_CAPTION? (And what does it mean exactly? I'm looking forward to the documentation for the different enum values to help me understand what they are and then we can better think about the names)